wgpu-rs/src/lib.rs

```diff
@@ -432,21 +455,19 @@ impl Device {
         }
     }
 
-    pub fn create_buffer_mapped<T>(&self, desc: &BufferDescriptor) -> (Buffer, &mut [T])
+    pub fn create_buffer_mapped<'a, T>(&self, desc: &BufferDescriptor) -> CreateBufferMapped<'a, T>
```

Comment:
what is the meaning of `'a` here? If you want it to be coming from `&self`, you can omit it completely. Otherwise it reads as it can give you *any* lifetime here, which is somewhat unsound

Reply:
It's the lifetime required to store an `&'mut [T]` in the return type. It's explicitly a synthetic lifetime because the reference is to a synthetic reference built from the mapped buffer.

Technically this entire construct isn't really legal in Rust, because `CreateBufferMapped` is self-referential (i.e. it stores a reference to interior state of the buffer it also contains)...

Reply:
I see: we can't elide the lifetime while we still have `T` argument that follows.